docs/cold_storage_test.py

```diff
@@ -0,0 +1,312 @@
+import subprocess
+import json
+import os
+from time import sleep
+from shlex import split, quote
+from s3_helpers import run_example, get_spec_path
+from datetime import datetime, timedelta, timezone
+# -
+
+# ## Exemplos
+
+
+# ### Trocar a classe de um objeto
+#
+# Para trocar a classe de um objeto para a classe de armazenamento cold storage, utilize o comando
+# mgc object-storage objects copy NOME_DO_BUCKET/NOME_DO_OBJETO MESMO_NOME_DO_BUCKET/MESMO_NOME_DO_OBJETO --storage-class=cold_instant
+
+# Para realizar o upload de um objeto com a classe de armazenamento cold storage, utilize o comando
+# mgc object-storage objects upload ARQUIVO NOME_DO_BUCKET --storage-class=cold_instant
+
+# + tags=["parameters"]
+config = "../params/aws-east-1.yaml"
+# -
+profile = config.split('/')[2].split('.')[0]
+
+LOGGER = logging.getLogger(__name__)
+
+# commands = [
+#     "mgc object-storage objects upload {object_key} {bucket_name} --storage-class=cold_instant",
+#     "aws --profile {profile} s3 cp {object_key} s3://{bucket_name} --storage-class=GLACIER_IR",
+#     "aws --profile {profile} s3api put-object --bucket {bucket_name} --key {object_key}",
+#     "rclone settier 'cold_instant' '{profile}:{bucket_name}/{object_key}'"
+# ]
+
+# # + {"jupyter": {"source_hidden": true}}
+# @pytest.mark.parametrize("cmd_template", commands)
+# def test_upload_object_with_cold_storage_class(cmd_template, empty_bucket_name):
+#     bucket_name = empty_bucket_name
+#     object_key = "file.txt"
+#     content = "Arquivo teste para upload com a classe de armazenamento Cold Storage"
+
+#     with open("file.txt", "w") as file:
+#         file.write(content)
+#         file.close()
+
+#     cmd = split(cmd_template.format(profile="br-se1", object_key=object_key, bucket_name=bucket_name, content=content))
+
+#     result = subprocess.run(cmd, capture_output=True, text=True)
+
+#     assert result.returncode == 0, f"Command failed with error: {result.stderr}"
+#     logging.info(f"Output from {cmd_template}: {result.stdout}")
+
+#     os.remove('file.txt')
+
+# run_example(__name__, "test_upload_object_with_cold_storage_class", config=config)
+
+def test_boto_upload_object_with_cold_storage_class(s3_client, empty_bucket_name):
+    # Configuração inicial
+    bucket_name = empty_bucket_name
+    object_key = "cold_file.txt"
+    content = "Arquivo de exemplo com a classe cold storage"
+
+    # Teste de upload
+    response = s3_client.put_object(
+        Bucket=bucket_name,
+        Key=object_key,
+        Body=content,
+        StorageClass="GLACIER_IR" 
+    )
+
+    response_status = response["ResponseMetadata"]["HTTPStatusCode"]
+    logging.info("Status do upload: %s", response_status)
+    assert response_status == 200, "Expected HTTPStatusCode 200 for successful upload."
+    
+    storage_class = s3_client.get_object_attributes(Bucket=bucket_name, Key=object_key, ObjectAttributes = ["StorageClass"])
+    
+    storage_class = storage_class.get("StorageClass")
```

Comment:
apenas uma preferencia pessoal minha, sinta-se livre para ignorar esta sugestão e marcar como resolvido, mas eu nao sou muito fã de reatribuir valor a mesma variavel...

```suggestion
    object_attributes = s3_client.get_object_attributes(Bucket=bucket_name, Key=object_key, ObjectAttributes = ["StorageClass"])
    
    storage_class = object_attributes.get("StorageClass")
```